samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -166,8 +166,10 @@ public ContainerProcessManager(Config config, SamzaApplicationState state, Metri
     }
 
     this.containerAllocator = new ContainerAllocator(this.clusterResourceManager, config, state, hostAffinityEnabled, this.standbyContainerManager);
-    this.allocatorThread = new Thread(this.containerAllocator, "Container Allocator Thread");
-    LOG.info("Finished container process manager initialization.");
+    if (shouldStartAllocateThread()) {
+      this.allocatorThread = new Thread(this.containerAllocator, "Container Allocator Thread");
```

Comment:
Do actually use containerAllocator anymore? Please walk through this part of the code with me offline.